### PR TITLE
Compiler detection

### PR DESCRIPTION
**Pull request by marehr**
See ticket #1349

### CODE REVIEW

File: util/cmake/FindSeqAn.cmake [hunk 1]
```diff
@@ -159,6 +158,45 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
...
+    # require at least clang 3.5
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
+        message(AUTHOR_WARNING "Clang version (${CMAKE_CXX_COMPILER_VERSION}) should be at least 3.5! Anything below is untested.")
+    endif()
+
+elseif(COMPILER_IS_MSVC)
+
+    # require at least MSVC 19.0
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "19.0")
+        message(FATAL_ERROR "MSVC version (${CMAKE_CXX_COMPILER_VERSION}) must be at least 19.0 (Visual Studio 2016)!")
```
Comment (rrahn) [inline util/cmake/FindSeqAn.cmake]: You mean Visual Studio 2015?

**Comment by marehr** [comment]
### Reason why the travis builds fails

The compiler version detection is done by the cmake variable `CMAKE_<LANG>_COMPILER_VERSION`. This feature is available since [cmake 2.8.8](https://cmake.org/cmake/help/v2.8.8/cmake.html#variable:CMAKE_LANG_COMPILER_VERSION).

After a short talk with hannes, we decided to require cmake 3.0. But it turns out that travis only offers cmake 2.8.7, see http://cdash.seqan.de/viewConfigure.php?buildid=160411.

So @h-2 what should we do?

**Comment by rrahn** [comment]
You should as well update the supported compilers on the README.

**Comment by rrahn** [comment]
Oh you did already. My bad!

File: util/cmake/FindSeqAn.cmake [hunk 2]
```diff
@@ -159,6 +158,45 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
...
+if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  set (COMPILER_IS_MSVC TRUE)
+endif (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+
+# ----------------------------------------------------------------------------
+# Check required compiler versions.
+# ----------------------------------------------------------------------------
+
+if(CMAKE_COMPILER_IS_GNUCXX)
+
```
Comment (h-2) [inline util/cmake/FindSeqAn.cmake]: before paranthesis ;)

**Comment by h-2** [comment]
after you introduce the space, I will merge it :)

**Pull request merged**